turbinia/evidence.py

```diff
@@ -958,3 +958,38 @@ def _postprocess(self):
       # Unmount the container's filesystem
       mount_local.PostprocessUnmountPath(self._container_fs_path)
       self.state[EvidenceState.CONTAINER_MOUNTED] = False
+
+
+#TODO implement support for several ewf devices if there are more than one
+#inside the block path
+class EwfDisk(RawDisk):
+  """Evidence object for a EWF based evidence.
+
+  Attributes:
+    device_path (str): Path to the mounted loop device.
+    ewf_path (str): Path to mounted EWF image.
+    block_path (str): Path to EWF mount directory.
+  """
+  POSSIBLE_STATES = [EvidenceState.ATTACHED, EvidenceState.MOUNTED]
+
+  def __init__(self, *args, **kwargs):
+    """Initialization for EWF evidence object."""
+    self.device_path = None
+    self.ewf_path = None
+    self.block_path = None
+    super(EwfDisk, self).__init__(*args, **kwargs)
+
+  def _preprocess(self, _, required_states):
```

Comment:
Regarding tests for these methods, some of the tests in this file are probably decent examples:
https://github.com/google/turbinia/blob/6e1b042c57381f02de13507a910c69340ea4bb99/turbinia/processors/mount_local_test.py#L149

And as joachim mentioned elsewhere, there are some E01 test files that we could use in the DfVFS repo that would could put into our `test_data` directory and use as well: https://github.com/log2timeline/dfvfs/tree/main/test_data